libs/geoservices/covid/src/lib/pages/maps/components/time-map/time-map.component.ts

```diff
@@ -0,0 +1,293 @@
+import { Component, OnInit } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { MapboxMapService } from '@tamu-gisc/maps/mapbox';
+import { Moment } from 'moment';
+import { forkJoin } from 'rxjs';
+
+@Component({
+  selector: 'tamu-gisc-time-map',
+  templateUrl: './time-map.component.html',
+  styleUrls: ['./time-map.component.scss']
+})
+export class TimeMapComponent implements OnInit {
+  constructor(private mapService: MapboxMapService, private http: HttpClient) {}
+
+  private maxDate: string;
+  private mortalButtonToggled: boolean;
+  private stateButtonToggle: boolean;
+  private selectedDate: string;
+  private stateData: object;
```

Comment:
Avoid the use of the basic `object` type.

The `object` type will not allow you to access members that are not part of the base object prototype because it assumes it's just that. It has no idea if the object is an array, and you will not receive any help from the editor with method suggestions.

```js
private stateData: object;

this.stateData.someProperty = "some value"; // Property 'someProperty' does not exist on type 'object'

this.stateData.forEach((s, i, a) => {
  // Logic
}) //Property 'forEach' does not exist on type 'object'
```

If you know the data structure, create interfaces. This will provide type-checking through your application and provide suggestions in template.

```js
interface IStateRecord {
    name: string;
    population: number;
}
```

```js
private stateData: Array<IStateRecord>;
```

Reply:
created interfaces for the county and state data.

Reply:
had some trouble with the array implementation so I'm just removing the 'object' type from the declaration